components/automatic-points-table/commons/display.lua

```diff
@@ -0,0 +1,243 @@
+
+local _POINTS_TYPE = {
+	MANUAL = 'MANUAL',
+	PRIZE = 'PRIZE',
+	SECURED = 'SECURED'
+}
+
+local PointsDivTable = Class.new(
+	function(self, pointsData, tournaments, positionBackgrounds)
+		self.root = mw.html.create('div') :addClass('divTable')
+			:addClass('border-color-grey') :addClass('border-bottom')
+
+		local columnCount = Array.reduce(tournaments, function(count, t)
+			return count + (t.shouldDeductionsBeVisible and 2 or 1)
+		end, 0)
+
+		local innerWrapper = mw.html.create('div') :addClass('fixed-size-table-container')
+			:addClass('border-color-grey')
+			:css('width', tostring(450 + (columnCount * 50)) .. 'px')
+			:node(self.root)
+
+		self.wrapper = mw.html.create('div') :addClass('table-responsive')
+			:addClass('automatic-points-table')
+			:node(innerWrapper)
+
+		self.rows = {}
+		self.pointsData = pointsData
+		self.tournaments = tournaments
+		self.positionBackgrounds = positionBackgrounds
+	end
+)
+
+local TableRow = Class.new(
+	function(self, pointsData, tournaments, positionBackground)
+		self.root = mw.html.create('div'):addClass('divRow')
+		self.cells = {}
+
+		local team = pointsData.team
+		if team.bg ~= nil then
+			self.root:addClass('bg-' .. team.bg)
+		end
+
+		self.pointsData = pointsData
+		self.tournaments = tournaments
+		self.team = team
+		self.positionBackground = positionBackground
+	end
+)
+
+local TableHeaderRow = Class.new(
+	function(self, tournaments)
+		self.tournaments = tournaments
+		self.root = mw.html.create('div') :addClass('divHeaderRow') :addClass('diagonal')
+		self.cells = {}
+	end
+)
+
+function PointsDivTable:row(row)
+	table.insert(self.rows, row)
+	return self
+end
+
+function PointsDivTable:create()
+	local headerRow = TableHeaderRow(self.tournaments)
+	self:row(headerRow)
+
+	Table.iter.forEachIndexed(self.pointsData, function(index, teamPointsData)
+		local positionBackground = self.positionBackgrounds[index]
+		self:row(TableRow(teamPointsData, self.tournaments, positionBackground))
+	end)
+
+	for _, row in pairs(self.rows) do
+		self.root:node(row:create())
+	end
+
+	return self.wrapper
+end
+
+function TableRow:create()
+	-- fixed cells
+	self:positionCell(self.pointsData.position, self.positionBackground)
+	self:nameCell(self.team)
+	self:totalCell(self.pointsData.totalPoints)
+	-- variable cells
+	Table.iter.forEachIndexed(self.pointsData, function(index, points)
+		local tournament = self.tournaments[index]
+		self:pointsCell(points, tournament)
+		if tournament.shouldDeductionsBeVisible then
+			self:deductionCell(points.deduction)
+		end
+	end)
+
+	for _, cell in pairs(self.cells) do
+		self.root:node(cell)
+	end
+	return self.root
+end
+
+local function wrapInDiv(text)
+	return mw.html.create('div'):wikitext(tostring(text))
+end
+
+function TableRow:baseCell(text, bg, bold)
+	local div = wrapInDiv(text) :addClass('divCell') :addClass('va-middle')
+		:addClass('centered-cell')	:addClass('border-color-grey') :addClass('border-top-right')
+
+	if bg ~= nil then

```

Comment:
Optional - effectively the same effect (`false` and `nil` are excluded with this)
```suggestion
	if bg then
```
Same goes on various other points in the codem, like line 52 & 234